testing_and_setup/compass/ocean/scripts/vertical_grid/make_vertical_grid.py

```diff
@@ -0,0 +1,198 @@
+    parser.add_argument(
+        '-maxit',
+        '--max_iterations',
+        dest='maxit',
+        default=1000,
+        help='maximum number of iterations for grid convergences',
+        type=int)
+    args = parser.parse_args()
+
+    create_vertical_grid(args.bottom_depth, args.nz, args.dz1,
+                         args.dz2, args.plot_vertical_grid,
+                         maxit=args.maxit, epsilon=args.epsilon,
+                         outFile=args.output_filename_name)
+# }}}
+
+
+def create_vertical_grid(
+        bottom_depth,
+        nz,
+        dz1_in,
+        dz2_in,
+        plot_vertical_grid,
+        maxit,
+        epsilon,
+        outFile):
+    # {{{
+    print('Creating mesh with ', nz, ' layers...')
+    dz1 = dz1_in
+    dz2 = dz2_in
+    # open a new netCDF file for writing.
+    ncfile = Dataset(outFile, 'w')
+    # create the depth_t dimension.
+    ncfile.createDimension('nVertLevels', nz)
+
+    refBottomDepth = ncfile.createVariable(
+        'refBottomDepth', np.dtype('float64').char, ('nVertLevels'))
+    refMidDepth = ncfile.createVariable(
+        'refMidDepth', np.dtype('float64').char, ('nVertLevels'))
+    refLayerThickness = ncfile.createVariable(
+        'refLayerThickness', np.dtype('float64').char, ('nVertLevels'))
+
+    Hmax = bottom_depth
+    nLayers = 0
+
+    layerThickness = np.zeros(nz)
+    dz = [epsilon]
+    z = [0]
+    count = 0
+
+    while nLayers != nz and count < maxit:
+        zval = -epsilon
+        dz = [epsilon]
+        z = [0]
+        nLayers = 0
+        while zval > -Hmax:
+            difference = dz_z(zval, Hmax, epsilon, dz2) - zval
+            while abs(difference) > 0.3:
+                zval -= epsilon
+                difference = dz_z(zval, Hmax, epsilon, dz2) - z[nLayers] + zval
+            z.append(zval)
+            dz.append(dz_z(zval, Hmax, epsilon, dz2))
+            nLayers += 1
+            zval -= epsilon
+
+        dz_arr = np.asarray(dz)
+        ind = abs(dz_arr - dz1).argmin()
+
+        dztemp = dz_arr[ind:]
+        nLayers = len(dztemp)
+        change = nz - nLayers
+        dz2 -= float(change)
+        count += 1
+
+    layerThickness[:nLayers] = dztemp
+    nVertLevels = nz
+    botDepth = np.zeros(nVertLevels)
+    midDepth = np.zeros(nVertLevels)
+    botDepth[0] = layerThickness[0]
+    midDepth[0] = 0.5 * layerThickness[0]
+
+    for i in range(1, nVertLevels):
+        botDepth[i] = botDepth[i - 1] + layerThickness[i]
+        midDepth[i] = midDepth[i - 1] + 0.5 * \
+            (layerThickness[i] + layerThickness[i - 1])
+
+    if count >= maxit:
+        print('Error: grid did not converge, adjust parameters')
+    else:
+        refBottomDepth[:] = botDepth
+        refMidDepth[:] = midDepth
+        refLayerThickness[:] = layerThickness[:nVertLevels]
+        ncfile.close()
+
+    if plot_vertical_grid:
+        import matplotlib
+        import matplotlib.pyplot as plt
+        matplotlib.use('Agg')
```

Comment:
Imports like this are considered poor practice.  This should be moved to the top of the script even if it is not always used.  Also, the `use('Agg')` call needs to happen before `plt` gets imported or it won't work right.

Reply:
Thanks. That's good to know.